From 8b53582aa07252a3603b71aae68f955a08d708e9 Mon Sep 17 00:00:00 2001
From: yuyue <yuyue@mprc.pku.edu.cn>
Date: Wed, 29 Feb 2012 13:20:45 +0800
Subject: [PATCH 311/641] UniCore64: move unflatten_device_tree() after
 setup_arch_memory()

unflatten_device_tree calls memblock_alloc to allocate some memory,
but some memblock cannot be used before setup_arch_memory, so we
have to move unflatten_device_tree after it.

Signed-off-by: yuyue <yuyue@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/devtree.c |    2 --
 arch/unicore64/kernel/setup.c   |    2 ++
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/unicore64/kernel/devtree.c b/arch/unicore64/kernel/devtree.c
index 714ca05..7e5d669 100644
--- a/arch/unicore64/kernel/devtree.c
+++ b/arch/unicore64/kernel/devtree.c
@@ -33,6 +33,4 @@ void __init setup_arch_devtree(char *cmdline)
 	of_scan_flat_dt(early_init_dt_scan_chosen, cmdline);
 	of_scan_flat_dt(early_init_dt_scan_root, NULL);
 	of_scan_flat_dt(early_init_dt_scan_memory, NULL);
-
-	unflatten_device_tree();
 }
diff --git a/arch/unicore64/kernel/setup.c b/arch/unicore64/kernel/setup.c
index 903f4d8..7183aa5 100644
--- a/arch/unicore64/kernel/setup.c
+++ b/arch/unicore64/kernel/setup.c
@@ -2,6 +2,7 @@
 #include <linux/init.h>
 #include <linux/notifier.h>
 #include <linux/string.h>
+#include <linux/of_fdt.h>
 
 #include <asm/setup.h>
 #include <asm/setup_arch.h>
@@ -54,5 +55,6 @@ void __init setup_arch(char **cmdline_p)
 	setup_arch_cpuinfo();
 	setup_arch_param(cmdline_p);
 	setup_arch_memory();
+	unflatten_device_tree();
 	setup_arch_resource();
 }
-- 
1.7.9.5

